src/index.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import ReactDOM from 'react-dom';
+
+const INITIAL_STATE = {
+  input: '',
+  todos: [],
+};
+
+function App() {
+  const [state, setState] = useState(INITIAL_STATE);
+
+  const handledTodoInput = (e) => {
+    const todoText = e.target.value;
+
+    setState((prevState) => ({
+      ...prevState,
+      input: todoText,
+    }));
+  };
+
+  const handleAddTodo = (e) => {
+    e.preventDefault();
+
+    const newTodo = {
+      id: state.todos.length ? state.todos[state.todos.length - 1].id + 1 : 1,
+      text: state.input,
+      done: false,
+    };
+
+    setState((prevState) => ({
+      todos: [...prevState.todos, newTodo],
+      input: '',
+    }));
```

Comment:
```suggestion
    setState({
      ...state,
      todos: [
       ...state.todos,
       newTodo,
      ],
      input: '',
    });
```

과 같이 정리해줄 수 있을 것 같아요.
`state` 는 거추장스러워서 destructuring 해서 사용해도 좋겠네요 ;)